### ISSUE TITLE
Regression: Unable to override `compare` after es6 refactoring

### ISSUE THREAD

**Issue by homj** [crcn/sift.js#155]
Since sift has been refactored to es6 & cleaned up (with 8b1f4b9), the previous feature of providing a custom `compare` function (added via #97) is broken.

Previously, the compare function was a property of `sift`, but has since been declared as an exported function which can't be overridden.

I'd suggest to change this back to the previous behavior, or allow to override the compare function by passing it via `options`. Additionally, I'd like to override the `comparable` function as well (e.g. do convert a date string to a `Date` instance), but this is independent from the issue described above and should be tracked in another issue.

**Comment by crcn** [comment]
Should be fixed in `8.4.0`. You now have the option to pass a custom `compare` function as an option. For example:

``javascript
    class Item {
      constructor(value) {
        this.value = value;
      }
      compare(other) {
        return other && this.value === other.value ? 0 : -1;
      }
    }

    const filter = sift(new Item("a"), {
      compare(a, b) {
        if (a instanceof Item) {
          return a.compare(b);  
        }
        return compare(a, b);
      }
    });

    const items = [new Item("a"), new Item("b"), new Item("a")];
    const results = items.filter(filter); 
```

☝️ let me know if this fixes the issue for you. 

> I'd like to override the comparable function as well (e.g. do convert a date string to a Date instance), but this is independent from the issue described above and should be tracked in another issue.

Just added a ticket for this here: #156

**Comment by crcn** [comment]
FYI #156 is fixed (adding custom `comparable`). Version for that is `8.5.0`.
